services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/TaggingAdapter.java

```diff
@@ -39,19 +39,21 @@ private TaggingAdapter() {
     public String adapt(Tagging tagging) {
         StringBuilder tagBuilder = new StringBuilder();
 
-        Tagging taggingClone = tagging.toBuilder().build();
+        if (!tagging.tagSet().isEmpty()) {
```

Comment:
Lets also do a null check here .

```java
if (tagging != null && !tagging.tagSet().isEmpty()) 
```

The null check is required to prevent below cases

```
Tagging tags = null;
        s3.putObject(PutObjectRequest.builder()
                                     .bucket(BUCKET)
                                     .key(key)
                                     .tagging(tags)
                                     .build(), RequestBody.empty());

```